### PR TITLE
Fix coords() function

### PR DESCRIPTION
**Pull request by atedstone**
`Raster.coords()` was returning incorrect y-coordinate grids. This was probably caused by an inconsistency in syntax between the old Georaster class from which this function was taken, vs new rasterio functionality.

* Method updated
* Test case added
* No test case for images with positive y axis resolution increment - I don't have any test images like this to hand, ideas welcome!

### CODE REVIEW

File: geoutils/georaster.py [hunk 1]
```diff
@@ -1020,18 +1020,17 @@ def coords(self, offset='corner', grid=True):
...
         assert offset in ['corner', 'center'], "ctype is not one of 'corner', 'center': {}".format(offset)
 
-        xmin, ymin, xmax, ymax = self.bounds
-        dx = list(self.transform)[0]
-        dy = list(self.transform)[4]
+        dx = self.res[0]
+        dy = self.res[1]
 
-        xx = np.linspace(xmin, xmax, self.width + 1)[::int(np.sign(dx))]
-        yy = np.linspace(xmin, xmax, self.height + 1)[::int(np.sign(dy))]
```
Comment (rhugonnet) [inline geoutils/georaster.py:1034]: looks like it was xmin/xmax instead of ymin/ymax here

Comment (rhugonnet) [reply geoutils/georaster.py:1034]: my mistake, probably... weird because I tested it at some point

**Review by rhugonnet — APPROVED** [review]

**Comment by rhugonnet** [comment]
looks great!

**Comment by rhugonnet** [comment]
I'm thinking: this modification will affect the current "interp_points" function where coords is used. There is no assertion yet in the test_interp() so the tests passed, but I'll have a look at the printed results of the test to see if it's still consistent or if we need to fix interp_points as well

**Comment by atedstone** [comment]
thanks @rhugonnet , i'll wait to hear back before we merge in this PR

**Review by adehecq — APPROVED** [review]
Sounds all good to me!
For the negative y resolution, we could create a synthetic test case.

**Comment by adehecq** [comment]
Want to merge this @atedstone ?

**Comment by atedstone** [comment]
@rhugonnet , do you want to look at `interp_points` before we close this PR?

**Comment by rhugonnet** [comment]
> @rhugonnet , do you want to look at `interp_points` before we close this PR?

Just opened #100 to check this in details, we can merge this PR in the meantime :+1:

**Pull request merged**